pallets/parachain-staking/src/lib.rs

```diff
@@ -1715,6 +1743,87 @@ pub mod pallet {
+        ) -> DispatchResult {
+            ensure_none(origin)?;
+            let voting_session = Self::get_growth_voting_session(&growth_id);
+            process_reject_vote::<T>(&voting_session, validator.account_id.clone())?;
+
+            Self::deposit_event(Event::<T>::VoteAdded {
+                voter: validator.account_id,
+                growth_id,
+                agree_vote: false,
+            });
+            // TODO [TYPE: weightInfo][PRI: medium]: Return accurate weight
+            Ok(())
+        }
+
+        #[pallet::weight(0)]
+        #[pallet::call_index(35)]
+        pub fn end_voting_period(
+            origin: OriginFor<T>,
+            growth_id: GrowthId,
+            validator: Validator<<T as avn::Config>::AuthorityId, T::AccountId>,
+            _signature: <T::AuthorityId as RuntimeAppPublic>::Signature,
+        ) -> DispatchResult {
+            ensure_none(origin)?;
+            //Event is deposited in end_voting because this function can get called from
+            // `approve_growth` or `reject_growth`
+            Self::end_voting(validator.account_id, &growth_id)?;
+
+            // TODO [TYPE: weightInfo][PRI: medium]: Return accurate weight
+            Ok(())
+        }
+    }
+
+    #[pallet::validate_unsigned]
+    impl<T: Config> ValidateUnsigned for Pallet<T> {
+        type Call = Call<T>;
+
+        fn validate_unsigned(_source: TransactionSource, call: &Self::Call) -> TransactionValidity {
+            if let Call::end_voting_period { growth_id, validator, signature } = call {
+                let growth_voting_session = Self::get_growth_voting_session(growth_id);
+                return end_voting_period_validate_unsigned::<T>(
+                    &growth_voting_session,
+                    validator,
+                    signature,
+                )
+            } else if let Call::approve_growth { growth_id, validator, approval_signature, signature } =
+                call
+            {
+                if !<Growth<T>>::contains_key(growth_id.period) {
+                    return InvalidTransaction::Custom(ERROR_CODE_INVALID_GROWTH_PERIOD).into()
+                }
+
+                let growth_voting_session = Self::get_growth_voting_session(growth_id);                
+                let eth_encoded_data = Self::convert_data_to_eth_compatible_encoding(&growth_id.period)
+                    .map_err(|_| InvalidTransaction::Custom(ERROR_CODE_INVALID_GROWTH_DATA))?;
+
+                return approve_vote_validate_unsigned::<T>(
+                    &growth_voting_session,
+                    validator,
+                    eth_encoded_data.encode(),
+                    approval_signature,
+                    signature,
+                )
+            } else if let Call::reject_growth { growth_id, validator, signature } = call {
+                let growth_voting_session = Self::get_growth_voting_session(growth_id);
```

Comment:
Why don't we need to check if the growth_id.period exists, as we did in the approve branch?

Reply:
Good point, added extra check.